src/index.ts

```diff
@@ -199,13 +204,28 @@ class MultinetAPI {
       text = await fileToText(data);
     }
 
+    let metadata;
+    if (columnTypes) {
+      const columns = Object.keys(columnTypes).map((column) => ({
+        key: column,
+        type: columnTypes[column],
+      }));
+
+      metadata = metadata || {};
+      metadata = {
+        ...metadata,
+        columns,
+      };
```

Comment:
I don't understand this section. When is metadata ever populated beforehand? Since you define it on line `207`, I think this produces the same result:

```suggestion
      metadata = { columns };
```

And I suppose if you want to be fancy you could replace the whole block with this:

```
    const metadata = columnTypes ? {
      columns: Object.keys(columnTypes).map((column) => ({
        key: column,
        type: columnTypes[column],
      }))
    } : undefined;
```

Reply:
I think I was thinking ahead to when other predefined metadata values might also flow through this, and I wrote it that way with that mindset. Your way is obviously better, so I'll take a closer look and accept your change after verifying that it works as intended.